packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,282 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
   }
 }
 
-/// A notification that an [GlowingOverscrollIndicator] will start showing an
-/// overscroll indication.
+/// A Material Design visual indication that a scroll view has overscrolled.
 ///
-/// To prevent the indicator from showing the indication, call [disallowGlow] on
-/// the notification.
+/// A [StretchingOverscrollIndicator] listens for [ScrollNotification]s in order
+/// to stretch the content of the [Scrollable]. These notifications are typically
+/// generated by a [ScrollView], such as a [ListView] or a [GridView].
+///
+/// When triggered, the [StretchingOverscrollIndicator] generates an
+/// [OverscrollIndicatorNotification] before showing an overscroll indication.
+/// To prevent the indicator from showing the indication, call
+/// [OverscrollIndicatorNotification.disallowIndicator] on the notification.
+///
+/// Created by [ScrollBehavior.buildOverscrollIndicator] on platforms
+/// (e.g., Android) that commonly use this type of overscroll indication when
+/// [ScrollBehavior.androidOverscrollIndicator] is true. Otherwise, the
+/// default [GlowingOverscrollIndicator] is applied.
+///
+/// See also:
+///
+///  * [OverscrollIndicatorNotification], which can be used to prevent the stretch
+///    effect from being applied at all.
+///  * [NotificationListener], to listen for the
+///    [OverscrollIndicatorNotification]
+class StretchingOverscrollIndicator extends StatefulWidget {
+  /// Creates a visual indication that a scroll view has overscrolled by
+  /// applying a stretch transformation to the content.
+  ///
+  /// In order for this widget to display an overscroll indication, the [child]
+  /// widget must contain a widget that generates a [ScrollNotification], such
+  /// as a [ListView] or a [GridView].
+  ///
+  /// The [axisDirection] and [notificationPredicate] arguments must not be null.
+  const StretchingOverscrollIndicator({
+    Key? key,
+    required this.axisDirection,
```

Comment:
I'm getting the error below if I set this property to `AxisDirection.left` or `AxisDirection.right`:

```dart
======== Exception caught by gesture ===============================================================
The following assertion was thrown while handling a gesture:
'package:flutter/src/widgets/overscroll_indicator.dart': Failed assertion: line 736 pos 14: 'notification.metrics.axis == widget.axis': is not true.
```

The code is (the listView uses `scrollDirection: Axis.vertical` by default):

```dart
StretchingOverscrollIndicator(
        axisDirection: AxisDirection.right,
        child: ListView.builder(
          itemCount: 200,
          itemBuilder: (context, index) {
            return Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text('${index * 1000}'),
            );
          },
        ),
      )
```

From my understanding, this should match the axis/scrollDirection of the inner scrollable widget, I believe this is woth documenting :)

Reply:
This is because the AxisDirection of a vertical scroll view is up or down, not left or right. This error is correct I think.